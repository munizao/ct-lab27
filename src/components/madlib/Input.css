.Input {
  color: #ffffff;
  background-color: #6b1a1a;
}

::placeholder {
  color: rgb(230, 78, 78);
  opacity: 1; 
}