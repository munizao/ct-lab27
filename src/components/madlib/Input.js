import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.css';

const Input = ({ placeholder, onChange, index, value }) => (
  <input className={styles.Input} value={value} name={'input' + index} type="text" onChange={onChange} placeholder={placeholder} />
);

Input.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired
};

export default Input;
