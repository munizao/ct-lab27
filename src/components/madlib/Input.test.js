import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input component', () => {
  it('renders Input', () => {
    const wrapper = shallow(<Input value={'wobble!'} key={1} placeholder={'wibble!'} onChange={() => {}} index={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
