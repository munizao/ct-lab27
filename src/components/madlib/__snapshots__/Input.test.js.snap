// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input 1`] = `
<input
  className="Input"
  name="input1"
  onChange={[Function]}
  placeholder="wibble!"
  type="text"
  value="wobble!"
/>
`;
